editor/editor_vcs_interface.cpp

```diff
@@ -418,8 +418,13 @@ void EditorVCSInterface::create_vcs_metadata_files(VCSMetadata p_vcs_metadata_ty
 		if (f.is_null()) {
 			ERR_FAIL_MSG("Couldn't create .gitattributes in project path.");
 		} else {
-			f->store_line("# Normalize EOL for all files that Git considers text files.");
+			f->store_line("# Properly detect languages on GitHub");
+			f->store_line("*.gd linguist-language=GDScript");
+			f->store_line("");
+			f->store_line("# Normalize EOL for all files that Git considers text files");
 			f->store_line("* text=auto eol=lf");
+			f->store_line("# Except for Batch files, which don't behave correctly otherwise");
+			f->store_line("*.bat eol=crlf")
```

Comment:
While this is indeed the case for `.bat` files, I don't know to what extent we want to include this for files that are possible but unlikely to be in a Godot project. We could also throw in `*.hdr binary` so that Git doesn't detect those images as text files, but again it's unlikely as most Godot projects don't use them. I don't have a strong opinion, either way works.

Reply:
The reason I put bat in there is because I believe Godot 4 has more multiplayer stuff out of the box, so one use case could be for running a server without editor without having to export project I guess?

I wasn't aware git mistakes .hdr for text files that's kind of strange, you'd think those cases would be solved in the modern age. But yea if there are more issues than these, maybe it's best off leaving them all out.

Reply:
Also, Windows users can write scripts using PowerShell `.ps1` files, which work with LF if I recall correctly.